jcg_testcases/src/main/scala/FileOperations.scala

```diff
@@ -0,0 +1,94 @@
+import org.apache.commons.io.FileUtils
+
+import java.io.{File, PrintWriter}
+
+object FileOperations {
+    private val pathSeparator = File.pathSeparator
+
+    /**
+     * Writes the given content to the given file and catches any exceptions.
+     *
+     * @param file    the file to write to
+     * @param content the content to write
+     */
+    def writeToFile(file: File, content: String): Unit = {
+        var printWriter: PrintWriter = null
+
+        try {
+            printWriter = new PrintWriter(file)
+            printWriter.write(content)
+        } catch {
+            case e: Exception => println(s"Error writing to file: ${e.getMessage}")
+        } finally {
+            if (printWriter != null) {
+                printWriter.close()
+            }
+        }
+    }
+
+    /**
+     * Returns all markdown files in the given directory.
+     *
+     * @param dir    the directory to search in
+     * @param filter a filter to apply to the file names
+     */
+    def listMarkdownFiles(dir: File, filter: String): Array[File] = {
```

Comment:
Maybe make this more generic, taking the file ending as a parameter? We might need this for other file types as well